src/backend/gpopt/translate/CTranslatorDXLToPlStmt.cpp

```diff
@@ -4724,6 +4728,16 @@ CTranslatorDXLToPlStmt::TranslateDXLTblDescrToRangeTblEntry(
 
 	rte->eref = alias;
 
+	/*
+		Store the alias information only when table alias is used to allow 
+		EXPLAIN find it
+	*/
+	if (optimizer_enable_table_alias && NULL != table_descr->MdAlias())
+	{
+		rte->alias = alias;
```

Comment:
Why is rte->eref equal to rte->alias in this case?

Reply:
I meant why do you change rte->eref? Maybe I shouldn't have done this?

Reply:
Still don't understand, I can't see where rte->eref is changed in this commit.

Reply:
so question: why do you change rte->eref while you are using rte->alias?

Reply:
In the parsenodes.h we can see comment:
>  *	  eref is the table reference name and column reference names (either
> *	  real or aliases).
And since we are to support aliases in Orca it seems logical to see actual alias in the eref.

Reply:
Then why don't you use rte->eref below?

Reply:
Why are you changing `rte->eref->aliasname`? Maybe it’s enough to just change `rte->alias->aliasname`?

Reply:
As I wrote rte->eref is a table reference name either real or alias. And now alias is supported and available nearby. I found no reason to use real name still. Regression shown that it is valid approach.

Reply:
I get it: are you using `rte->alias` below as `NULL != table_descr->MdAlias()` condition?